packages/expo-permissions/src/PermissionsHooks.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { PermissionType, PermissionResponse, askAsync, getAsync } from './Permissions';
+
+/**
+ * Get or ask permission for protected functionality within the app.
+ * It returns the permission response after fetching or asking it.
+ * The hook fetches the permissions when rendered, by default.
+ * To ask the user permission, use the `askPermission` callback or `ask` option.
+ *
+ * @see https://docs.expo.io/versions/latest/sdk/permissions/
+ * @example const [permission, askPermission, getPermission] = usePermissions(...);
+ */
```

Comment:
As you're provided `@example` part, why not to provide fully-featured example? I believe those are the best 😊 
Like in this [TS playground](https://www.typescriptlang.org/play/index.html#code/PQKhCgAIUhxBTALgZ0gewE6QIbINaoBmmkADvBgLYCWyy1aAdgBTICUkxWpGai8AY34ATTgFdGQho2wAbaogCekAO4KAFtUaRE6+DlKkAdFBgAlJGIyNUu-eSq16TSBnjJSTZPuyF+WQiQBTUYAc3QsXDwtcIUTaEgAIWVheEJsMVlEHT1IdTQ0PE4gvVtchxo6aVQVPW03RlS3YXiYABU0HHwc-TFvAJIKp2kAGkg+-TtIAAMogAUKSudGacgBOVkAI2wBIpJZ-FW0UkRpVtNIAAFvfXVERFJkAC5gYGE0AWQjeAAPTyMGMAAG4UZbIYCybD8ZCIYDIYR4YBDKpeYAXSAXS6-bCUUiyeAXaZElA-C6ECRSFwAQUMzA4AG90WsvNkANrI5ZjKIAMUwC0cKMYY1CSH5S2kAF1IABecbeMXDLzMVkKwVfADCVIAsgBRMxUsaqsFGAAyAHlNW0AJJmgByErG9K6eCeOgwYn0AF82ABuC5M6iESDMACEHOkkAAPpGyItFYwjDCoX1ICHpbKAOShDDYRgiDMM0wASEgpbciCs2mYTNLpYAPAA1ajwFQAPhrtfrbV+iFbRojtEgjD4kGzuZEdeA3Z+vY7nbriTE9xcp0Q+OlACJYDm87GBcsN+hGHM3HRpfT5nHBZ7IMB2wlO-XgE2W-eYJ22EzPf6H5AjP+Lm-BIiWmUw0V+TwMGyclJFOFwJn7JUoB0RRyFdRDGDaVD9BjDCsPIVkHWQ444JsdCrzBM0TmqGVIHpb82FdFUKOkCwPC8HDxkaNItHgYQxjpGVW0gE80EqeA6yBNBqGEVsBI4aVhNE8TJOk2SpUZUsBBZSBWWEKFsDGbxEAAEQMqVZQmABlRAoQkjC2M8Gx4FbOk-S0nSnSiWj0lkbxhSQWjEHdL1aJI6p3OZGxsiUchUFlKkMBzRQAWQRLkuYWL4A4AB+FDyEgJisolP1kNeIc+HgV0FFQLR+DzaQ5B0Tp5BheBtEQTppiy6YxmHbJuuw5BpniW9gFUOoZh6yBB3WNrRE6nBtGwJLsEUMYFDINAqk2fE1jcOzUGwIcWyPfQXHgEEMGUBomlG8rFp4S72uyCRGEEdxkBW5QYMpbQtCTSR3FUfR3r45rIFa+qIamDccxUQ8gTkD0TGQ7ToudDD4rleB1Q2bZdmYQTFOdKlkEUSRmH-IwsvYGm6lYJAzNstgxlZYrfTGyB3HkPMAFphFobBdvgPnef0A6hD5-JCnBX51AyGFqBBAX4EecA0Z0kVECx2iJjx2Qth2PAiYU4TtbJimBCp-9abYen2sZ0yDNZ3SOZ9LmebqgWhZFsXeNceAdkQaWCgIYB5cV04VdSdXkImHVCECIRTaEujkNLQNgyiQtH0xlilV9DPIG-Yus9DbyADJK9HJBc8fbWsbc4ugJLtmoi5fAsYCxAe6xiUi+Q8tK10-TbM7vBu9rnWC5sErwFLiDMGyOqKHSAR9CxqjSNQTTbzASArSDTbkHyTJRG8qZwy8Cb2sgSg0AkEQxlSdJMhQCHpl87xVhANFSyiDlV0mwCj4lzJFUAMAj4zWyKfR+shRCBEQMEGajAuCUChBGYWj8YrlFnjUSaD8n58RfmkDIWRbBdWCh6X+-9p5AMgCAtAYDGB+k9EAA)? What would you say? 🤔

Reply:
Maybe we could link to Snack instead? That way, the user has a full testing environment that he or she can run on devices or web. I can add an embedded snack to the docs, and link it here. Would that be better? 😄

Reply:
I like the idea 👍, but I would rather have an inline example that I can copy/paste from 😉 
It's my long-term dream to have a small example next to each of the methods where I can see just enough info how to use it whenever I hover over them 😊 
<img width="832" alt="Screenshot 2020-06-16 at 10 47 29" src="https://user-images.githubusercontent.com/16623003/84753052-cf9c5e80-afbe-11ea-9937-f5a096fb42be.png">

Reply:
I believe having a snack for each example will cause too much overhead.. 😞 Unless we create some tool that would work just like `TS playground` - have the code inlined into URL 🤔
Or even better: have a processing tool that would grab this minimalistic example from `tsdoc` and autogenerate this snack URL 🥳

Reply:
haha yeah, I feel the same way. Having a quick reminder of how you can use the method is awesome DX. But I think there is a tradeoff related to how much info you provide. 

To give you a bit more context on why I chose a simple 1 liner as example:
The way I see it, we have all kinds of users on a scale from _beginner_ to _experts_. A full example would be awesome for the _beginner_, but would be too much for the _expert_ I think. In this case, I mostly focused on the hook-output because the types are really not describing this at all. Having a single or 2 lines example on what the output of the hooks are, helps both the _beginner_ and shouldn't be too much for _experts_. To still help the _beginner_, I added a link to the docs that contains a full example (so should be only 1 click away :))

That being said, I'm definitely open to this! I think we should trim this as much as possible, don't think we need the options in the example. Both the argument type as well as the actual options interface should be clear enough for both users (imho) 😄 And we might want to use `permissions?.status !== 'granted'`, stuff like that 😄

Reply:
Maybe [something like this](https://www.typescriptlang.org/play/index.html#code/PQKhCgAIUhxBTALgZ0gewE6QIbINaoBmmkADvBgLYCWyy1aAdgBTICUkxWpGai8AY34ATTgFdGQho2wAbaogCekAO4KAFtUaRE6+DlKkAdFBgAlJGIyNUu-eSq16TSBnjJSTZPuyF+WQiQBTUYAc3QsXDwtcIUTaEgAIWVheEJsMVlEHT1IdTQ0PE4gvVtchxo6aVQVPW03RlS3YXiYABU0HHwc-TFvAJIKp2kAGkg+-TtIAAMogAUKSudGacgBOVkAI2wBIpJZ-FW0UkRpVtNIAAFvfXVERFJkAC5gYGE0AWQjeAAPTyMGMAAG4UZbIYCybD8ZCIYDIYR4YBDKpeYAXSAXS6-bCUUiyeAXaZElA-C4CLzZADayOWY3mi2GTAAupAALzjbwLRwomzMLlLapGADCAEEALIAUTMIrYAG4Lhc3IgrNoadIAPxGUIYbCMEToyDqyAAHiFOIo2EgRmtkGAAD4DU8TYkxPcXEw5m46KyAN707nLAC+trt8oSROmpjRv08GGyhAkUhcE35jN5UB0inITtTPLaWf0AB9ILnlvnyJSmSMM8dTl4cwyecgAPInapsyA+wPgNhOyml6QWDxeIvjRppLTwYRjZgcVl2ku8SrwY1AtDUYR2mdzheetDL1frzcsn0Z8k2KnCKHYMbeRAAEWvLPZEwAyogoSuB0wh54bPA7VnMM1gpTsuiKdl0lkbwxlCJAO0QDAxH0YN2VrapgPPGFM3IVB2RFDAdUUAFkAIojmCUcgOCNSj9D7WimXlDNXkgRg+HgJ0FFQLR+D1aQ5B0Tp5BheBtEQTpplo6YxjY7JJILZBpniW1gFUOoZikyBaDWXAREEnBtGwQjsEUMYFDINAqk2fE1jcT9UEtRh4BUdAnNcyB4BBDBlAaJplJY8SyDcEE9THJyBHcZAjOUBNJDrbQtBhXUIpqfQnKnfThN4-SpgAIh1FRcsgIE5GQkwz1A-0BS8DsJjNWQth2PBmFnNkFyiEVkEUSRmGtIxaPYfq6lYJBHw-NgxkpBi5RUjzkHkPUAFphFobBrPgRaFv0OyhEW-JCnBX51AyGFqBBZb4EecAKovSA4MQb8bFq7x6sa3YWp3O6kE67qBF660BrYIbRJGh9rwmyApoLJkZpY9wFsQZbVvWzbJ1ceAdkR-aCGAI6TtOc7UiujMJglQhAiED62s7DNIC0whIGYKIOFPOm2fAx7kCA2nIG7HnqAZ5gAEIokgAAyMWvsQFmebp+7Oe59nuzpwNJqiOl8E52CkG1h7GzBGGmLppUVUhq8Pw1vAtalznGPAPmY0wbIeIodIIpLfXqlbeLUFZ20wEgABJBnzOQfJMlEUWpjVGralEyBKDQCQRDGVJ0kyFB9OmKDvFWEA0TpqJ1SdTYCnxXVgNAGBg607Iw6T2RRECRBgi0xguEoKFpBwUvXR6MhPdj9TE+TqdU7SDIslsCTEOQvOC6l4vIFLtBy8YeVAyAA)? A stripped out version of yours, but should still be an example.

Reply:
Added the stripped-down example 😄